src/simularium/types.ts

```diff
@@ -209,3 +212,27 @@ export interface CacheNode {
     next: CacheNode | null;
     prev: CacheNode | null;
 }
+
+export interface BaseSimulatorParams {
+    fileName: string;
+}
+
+export interface RemoteSimulatorParams extends BaseSimulatorParams {
+    netConnectionSettings: NetConnectionParams;
+    requestJson?: boolean;
+    prefetchFrames?: boolean;
+}
+
+export interface LocalProceduralSimulatorParams extends BaseSimulatorParams {
+    clientSimulatorImpl: IClientSimulatorImpl;
+}
```

Comment:
I would keep the names consistent with the class names for now, ie this should be `ClientSimulatorParams`. 
I think `ClientSimulator` makes sense as a class name given `RemoteSimulator`, I think the `clientSimulatorImpl` should be renamed. Some suggestions: `simulator` or `generator`